### ISSUE TITLE
Object Types Editor does not auto-reload when Object Types File changes on disk

### ISSUE THREAD

**Issue by vchelaru** [bjorn/tiled#1816]
Reproduced with:

Operating System: Windows 10
Tiled Version: 1.0.3 (32 bit)

To reproduce:

1. Create an XML file to contain object types (see below for sample file)
1. Open Tiled
1. Go to **View** -> **Object Types Editor**
1. In the Object Types Editor window, select **File** -> **Choose Object Types File**
1. Browse to the location where the .xml file is saved and select it
1. Verify that the Type appears, and when selected, the Properties show correctly
1. Open the XML file on disk
1. Add a new property
1. Save the file
1. Switch back to Tiled

Observe - the newly added (or removed) properties do not show up in the Object Types Editor. Similarly, closing and re-opening the Object Types Editor window does not refresh the properties from disk. However, closing and re-opening Tiled does result in the properties refreshing.
Expected - Tiled should have a 'file watch' for the XML file. When the file changes, the dialog should reload and display all new Types and Properties.

Why this is important: The object types xml may be checked in to source control and may be modified while Tiled is open. Furthermore some tools (such as FlatRedBall) generate this file while the user is working on their game. The user may be confused as to why new types or properties are not showing up despite the file having been generated.

Example file:

```
<?xml version="1.0"?>
<objecttypes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <objecttype name="Monster">
  </objecttype>
</objecttypes>
```

Modified example file:
```
<?xml version="1.0"?>
<objecttypes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <objecttype name="Monster">
    <property name="HP" type="int" />
  </objecttype>
</objecttypes>
```

Screenshot:

![nopropertyrefresh](https://user-images.githubusercontent.com/729631/32686748-84ffdc68-c668-11e7-9bae-d5fe5be1a56d.png)

**Comment by bjorn** [comment]
Hey there! I definitely agree it should auto-reload. Currently Tiled only watches explicitly opened files, tileset images and Python scripts, but it is lacking watchers for:

* Images in image collection tilesets (#990)
* External tilesets that aren't explicitly opened (#1785)
* Image layer images (#825)
* Object Types (this issue)
* Object Templates
* Tile Stamps

Implementing some of these will be non-trivial, but watching and reloading of the object types file in particular should be relatively easy to implement, at least. Thanks for opening an issue about it!

**Comment by bjorn** [comment]
Closing this issue since it was implemented by #2674. The other things that could be auto-reloaded have their own issues associated with them (except for tile stamps, need to check first if they really need it).
